sdk/src/main/java/com/horizen/transaction/DefaultTransactionIncompatibilityChecker.java

```diff
@@ -42,20 +42,23 @@
     }
 
     private <T extends BoxTransaction> boolean checkOutputBoxesPositionExclusivity(T newTx, List<T> currentTxs) {
-        List<Long> currentTxsOutputBoxesPosition = currentTxs.stream().flatMap(tx -> {
+        List<Long> currentTxsOutputBoxesPositionList = currentTxs.stream().flatMap(tx -> {
             List<Box> boxes = tx.newBoxes();
             return boxes.stream().map(box -> tree.getPositionForBoxId(new ByteArrayWrapper(box.id())));
         }).collect(Collectors.toList());
 
-        List<Box> newTxsBoxes = newTx.newBoxes();
-        List<Long> newTxOutputBoxesPosition = newTxsBoxes.stream().map(box -> tree.getPositionForBoxId(new ByteArrayWrapper(box.id()))).collect(Collectors.toList());
+        Set<Long> currentTxsOutputBoxesPositionSet = new HashSet<>(currentTxsOutputBoxesPositionList);
+        if (currentTxsOutputBoxesPositionList.size() != currentTxsOutputBoxesPositionSet.size()) {
```

Comment:
this case means, that mempool was inconsistent even before. Should never happen. Let's assert here.

Reply:
done